src/nvim/edit.c

```diff
@@ -1846,47 +1848,49 @@ static void ins_ctrl_x(void)
   }
 }
 
-/*
- * Return TRUE if the 'dict' or 'tsr' option can be used.
- */
-static int has_compl_option(int dict_opt)
+/// Check that the "dict" or "tsr" option can be used.
+///
+/// @param  dict_opt  Check "dict" when true, "tsr" when false.
+static bool has_compl_option(bool dict_opt)
 {
-  if (dict_opt ? (*curbuf->b_p_dict == NUL && *p_dict == NUL
-                  && !curwin->w_p_spell
-                  )
+  if (dict_opt ? (*curbuf->b_p_dict == NUL
```

Comment:
It is not stricly required by the style guide, but it would be nice to align the `?` with the `:` whenever reasonable (here you can put `?` on a new line and join with the next)